### PR TITLE
chore: missing typing for AC

### PR DESCRIPTION
**Pull request by chemelli74**

### CODE REVIEW

File: midealocal/devices/ac/__init__.py [hunk 1]
```diff
@@ -334,7 +336,7 @@ def set_attribute(self, attr: str, value: Any) -> None:
                     )
                     setattr(message, str(self._fresh_air_version), fresh_air)
             elif attr in self._attributes.keys():
-                message = self.make_message_uniq_set()
+                message = self.make_message_set()
```
Comment (Necroneco) [inline midealocal/devices/ac/__init__.py:339]: Is this safe?

File: midealocal/devices/ac/__init__.py [hunk 2]
```diff
@@ -354,15 +356,17 @@ def set_attribute(self, attr: str, value: Any) -> None:
...
+        )
         message.target_temperature = target_temperature
         if mode is not None:
             message.power = True
             message.mode = mode
         self.build_send(message)
 
     def set_swing(self, swing_vertical: bool, swing_horizontal: bool) -> None:
-        message: MessageGeneralSet = self.make_message_uniq_set()
+        message: MessageGeneralSet = self.make_message_set()
```
Comment (Necroneco) [inline midealocal/devices/ac/__init__.py:369]: And this

**Review by Necroneco — CHANGES_REQUESTED** [review]

File: midealocal/devices/ac/__init__.py (hunk 1 above)
Comment (chemelli74) [reply midealocal/devices/ac/__init__.py:339]: Yes, following the code `make_message_uniq_set()` it just calls 2 different methods based on a flag.
One of the 2 doesn't have the needed fields here, so...

File: midealocal/devices/ac/__init__.py (hunk 2 above)
Comment (chemelli74) [reply midealocal/devices/ac/__init__.py:369]: Same as above

File: midealocal/devices/ac/__init__.py (hunk 1 above)
Comment (Necroneco) [reply midealocal/devices/ac/__init__.py:339]: But the following code with needed fields is in `if` block...

File: midealocal/devices/ac/__init__.py (hunk 2 above)
Comment (Necroneco) [reply midealocal/devices/ac/__init__.py:369]: Here can assume ok

File: midealocal/devices/ac/__init__.py (hunk 1 above)
Comment (chemelli74) [reply midealocal/devices/ac/__init__.py:339]: You need a `MessageGeneralSet` object or it won't have the needed fields.
And the original method returns `MessageSubProtocolSet | MessageGeneralSet` based on the if.

So is safe to use always the method that returns  `MessageGeneralSet`

Comment (Necroneco) [reply midealocal/devices/ac/__init__.py:339]: If not enter the following `if`,  `MessageSubProtocolSet` seems also ok?

Comment (chemelli74) [reply midealocal/devices/ac/__init__.py:339]: Nope.
Check the following fields:

image

those are available only for `MessageGeneralSet` ;-)

Comment (Necroneco) [reply midealocal/devices/ac/__init__.py:339]: Yes, I mean, if `self._used_subprotocol` is `True` and `attr` not in these attributes,  is this possible?

Comment (Necroneco) [reply midealocal/devices/ac/__init__.py:339]: I didn't go too deep into the code, just asked questions about this control flow.
If this situation is impossible, then there is no problem.

Comment (Necroneco) [reply midealocal/devices/ac/__init__.py:339]: Ok, I can give approval, but I still think this could be a change

Comment (Necroneco) [reply midealocal/devices/ac/__init__.py:339]: https://github.com/rokam/midea-local/pull/33#issuecomment-2254546944

**Pull request merged**